statistics: Added note about no TopN collected for unique single columns (#22475)
pingcap/docs: statistics.md

TiDB records the values and occurrences of Top-N values. Here `N` is controlled by the `WITH NUM TOPN` parameter. The default value is 20, meaning the top 20 most frequent values are collected. The maximum value is 1024. For details about the parameter, see [Manual collection](#manual-collection).

> **Note:**
>

> > When [`tidb_analyze_version = 2`](/system-variables.md#tidb_analyze_version-new-in-v510), TiDB does not collect Top-N statistics for single-column unique indexes or for their corresponding columns, because each value can occur at most once.

Reviewer: ```suggestion
> Starting from v9.0.0, when [`tidb_analyze_version = 2`](/system-variables.md#tidb_analyze_version-new-in-v510), TiDB does not collect Top-N statistics for single-column unique indexes or for their corresponding columns, because each value can occur at most once.
```

> > When [`tidb_analyze_version = 2`](/system-variables.md#tidb_analyze_version-new-in-v510), TiDB does not collect Top-N statistics for unique indexes that contain only a single column, or for the corresponding column statistics. This is because each value can occur at most once.

Reviewer: ```suggestion
> When [`tidb_analyze_version = 2`](/system-variables.md#tidb_analyze_version-new-in-v510), TiDB does not collect Top-N statistics for single-column unique indexes or for their corresponding columns, because each value can occur at most once.
```
